multiqc/multiqc.py

```diff
@@ -689,8 +690,8 @@ def run(
                 except AttributeError:
                     pass
 
-        except UserWarning:
-            logger.debug("No samples found: {}".format(list(mod_dict.keys())[0]))
+        except (ModuleNoSamplesFound, UserWarning):  # UserWarning deprecated from 1.16
+            logger.debug(f"No samples found: {this_module}")
```

Comment:
Let's throw a debug log message here about the depreciation, maybe use `warnings.DepreciationWarning` ?
```suggestion
        except ModuleNoSamplesFound:
            logger.debug(f"No samples found: {this_module}")
        except UserWarning:  # UserWarning deprecated from 1.16
            logger.debug(f"DEPRECIATED: Please use ModuleNoSamplesFound instead of UserWarning")
            logger.debug(f"No samples found: {this_module}")
```

Actually could also use use `config.lint` here so that we throw an error when running with `--lint`. That would be good to catch this in any new modules etc.

Reply:
Tried with `DeprecationWarning`, it's a bit ugly, and running python with `-W error` also fails on warnings from the library code which is not what we want.

But we already have the mechanism for handling this, as you said: `config.lint` and `report.linting_errors`. Implemented it this way instead.

Reply:
Ok, sounds good! 👍🏻